### PR TITLE
Balance changes for Chronotrain

### PR DESCRIPTION
**Pull request by danielmartin0**
in order to provide more distinction between the difficulty levels, this pull request contains the balance modifications:
- coins required for upgrades scales with difficulty
- manual mining speed scales with difficulty
- distribution of ore richness on planets changes somewhat with difficulty
- below normal difficulty, the pollution absorbed by sending biters now scales with the inverse square of the difficulty rather than inverse-linearly. The pollution threshold for attacks scales the same way. Therefore the strength of biter attacks on each difficulty are changed as Piece of Cake->Easy, Peaceful->Piece of Cake.

In addition some small balance bugs have been fixed.

Needs checking that I haven’t introduced bugs inadvertently (especially around the code to update mining speed)

### CODE REVIEW

**Comment by rocifier** [comment]
I would like to see some of this information reflected in the map description or the difficulty selection dialog. I think people are generally unaware what picking a particular difficulty does, and unless they are a developer they can't check.

**Comment by danielmartin0** [comment]
Agreed

**Comment by hanakocz** [comment]
I definitely have difficulty vote gui rework on my todo list!

File: maps/chronosphere/event_functions.lua [hunk 1]
```diff
@@ -286,22 +286,35 @@ local mining_researches = {
...
+	if difficulty > 1 then
+		manual_mining_difficulty_scaling = 1.8 - ((difficulty - 1) / 3)
+	end
+
+
 	if event == nil then
 		-- initialization/reset call
-		game.forces.player.mining_drill_productivity_bonus = game.forces.player.mining_drill_productivity_bonus + 1
-		game.forces.player.manual_mining_speed_modifier = game.forces.player.manual_mining_speed_modifier + 1
+		game.forces.player.mining_drill_productivity_bonus = 1 --fixed typo?
```
Comment (hanakocz) [inline maps/chronosphere/event_functions.lua:303]: Minin productivity starts at +100% , intentionaly. Also, it is possible that the initial mining productivity is non-zero (for example from installed mod), thus why it is not hard-written 1

File: maps/chronosphere/event_functions.lua [hunk 2]
```diff
@@ -312,8 +325,8 @@ end
 function Public_event.on_technology_effects_reset(event)
 	local objective = Chrono_table.get_table()
 	if event.force.name == "player" then
-		game.forces.player.character_inventory_slots_bonus = game.forces.player.character_inventory_slots_bonus + objective.invupgradetier * 10
-		game.forces.player.character_loot_pickup_distance_bonus = game.forces.player.character_loot_pickup_distance_bonus + objective.pickupupgradetier
+		game.forces.player.character_inventory_slots_bonus = objective.invupgradetier * 10 --fixed typo?
+		game.forces.player.character_loot_pickup_distance_bonus = objective.pickupupgradetier --fixed typo?
```
Comment (hanakocz) [inline maps/chronosphere/event_functions.lua:329]: Same here, the previous state needs to be taken into account.

File: maps/chronosphere/upgrade_list.lua [hunk 3]
```diff
@@ -214,7 +222,7 @@ function Public.upgrades()
       tooltip = {"chronosphere.upgrade_computer3_tooltip"},
       jump_limit = 25,
       cost = {
-        item1 = {name = "coin", tt = "item-name", sprite = "item/coin", count = 2000},
+        item1 = {name = "coin", tt = "item-name", sprite = "item/coin", count = 20000 * upgrade_coin_cost_scale}, --typo fix on coin cost I think?
```
Comment (hanakocz) [inline maps/chronosphere/upgrade_list.lua:225]: This upgrade has 10 tiers, so each costs 2000 -> for final 20 000

File: maps/chronosphere/event_functions.lua (hunk 1 above)
Comment (danielmartin0) [reply maps/chronosphere/event_functions.lua:303]: I understood that it starts at +100%, but I didn't think about the mods. Thanks!

Btw, I am making some more changes, along with providing the balance changes as tooltips in the difficulty poll.

File: maps/chronosphere/event_functions.lua (hunk 2 above)
Comment (danielmartin0) [reply maps/chronosphere/event_functions.lua:329]: Yep.

File: maps/chronosphere/upgrade_list.lua (hunk 3 above)
Comment (danielmartin0) [reply maps/chronosphere/upgrade_list.lua:225]: Gotcha

**Comment by danielmartin0** [comment]
Ending updates to this pull request here. It contains balance changes which can be playtested and experimented with; other changes are cosmetic.

Working on various other game improvements locally.

**Comment by danielmartin0** [comment]
Apologies for so many updates earlier if people got notifications from them. Newbie here.

Making a separate pull request soon for significantly more content.

**Pull request closed without merging**